File: ABC/ABC142/E.py
def solve():
    N, M = map(int, input().split())
    INF = 10 ** 9
    
    dp = [INF] * (2 ** N)
    dp[0] = 0
    for _ in range(M):
        a, b = map(int, input().split())
        c = list(map(lambda x: int(x) - 1, input().split()))
        open = 0
        for i in range(b):
            open += 1 << c[i]
        for i in range(2 ** N):
            if dp[i] + a <= dp[i | open]:
                dp[i | open] = dp[i] + a
    
    if dp[-1] == INF:
        print(-1)
    else:
        print(dp[-1])


if __name__ == '__main__':
    solve()

File: ABC/ABC142/E.nim
import strutils, sequtils, math, algorithm


proc solve() =
    var
        N, M: int
        a, b: int
    (N, M) = stdin.readLine.split.map(parseInt)
    const INF = 10^9
    
    var dp: array[2^12, int]
    dp.fill(INF)
    dp[0] = 0

    for _ in 0..<M:
        (a, b) = stdin.readLine.split.map(parseInt)
        var
            c = stdin.readLine.split.map(parseInt)
            can_open = 0
        for ci in c:
            can_open += 1 shl (ci - 1)
        for i in 0..<2^N:
            dp[i or can_open] = min(dp[i or can_open], dp[i] + a)
    
    if dp[2^N - 1] == INF:
        echo -1
    else:
        echo dp[2^N - 1]


when is_main_module:
    solve()
